src/frontend/src/app/AppAuthenticated.tsx

```diff
@@ -42,6 +42,12 @@ const AppAuthenticated: React.FC<AppAuthenticatedProps> = ({ userId, userRole })
   const theme = useTheme();
   const [drawerOpen, setDrawerOpen] = useState(false);
   const [moveContent, setMoveContent] = useState(false);
+  const [clickedFinishline, setClickedFinishline] = useState(false);
+  const [isGuestOnMemberPage, setIsGuestOnMemberPage] = useState(false);
+
+  useEffect(() => {
+    setIsGuestOnMemberPage(isGuest(userRole) && clickedFinishline);
```

Comment:
You should almost never use a set state in a useEffect. This basically causes the page to rerender infinitely